node/src/components/block_validator.rs

```diff
@@ -48,8 +48,8 @@ pub enum Event<T, I> {
     DeployFound(DeployHash),
 
     /// A request to find a specific deploy, potentially from a peer, failed.
-    #[display(fmt = "deploy {} missing", _0)]
-    DeployMissing(DeployHash),
+    #[display(fmt = "deploy {} missing/invalid {}", _0, _1)]
+    DeployMissing(DeployHash, bool),
```

Comment:
Why not have a new `DeployInvalid` variant?

Reply:
I can do that, sure :) I tried to have a minimal set of changes too much.

Reply:
Done in [`37fd3f7` (#882)](https://github.com/CasperLabs/casper-node/pull/882/commits/37fd3f7ead882e9da4856cdb8e930d2a3444f674)